src/sagemaker/session.py

```diff
@@ -4196,6 +4202,21 @@ def _deployment_entity_exists(describe_fn):
         return False
 
 
+def _endpoint_config_name_exists(describe_fn):
+    """Checks if similar endpoint configuration name exists."""
+    try:
+        describe_fn()
+        return True
+    except ClientError as ce:
+        error_code = ce.response["Error"]["Code"]
+        if not (
+            error_code == "ValidationException"
+            and "Similar endpoint configuration name exists" in ce.response["Error"]["Message"]
```

Comment:
Is this the exact error message? Most List* apis just returns empty list if there is nothing matching.